src/python/dxpy/utils/executable_unbuilder.py

```diff
@@ -193,7 +226,30 @@ def untar_strip_leading_slash(tarfname, path):
 
                 untar_strip_leading_slash(fname, "resources")
                 os.unlink(fname)
-                deps_to_remove.append(dep)
+                # add dep name to deps_downloaded set
+                deps_downloaded.add(dep.get("name"))
+                
+            except DXError:
+                print("Download failed on dependent file with ID {} from region {}. Will try the next permitted region.\n".format(handler_id, region),
+                        file=sys.stderr)
+                # clean up deps already downloaded
+                deps_downloaded.clear()
+                created_resources_directory = False
```

Comment:
Files we already downloaded will be removed. I think it is better if all resources are retrieved from the same region just in case of version conflicts etc.